ui/js/compareperf.js

```diff
@@ -5,13 +5,46 @@
 "use strict";
 
 perf.controller('CompareChooserCtrl', [
-  '$state', '$stateParams', '$scope', 'ThRepositoryModel', 'ThResultSetModel',
-  function CompareChooserCtrl($state, $stateParams, $scope,
+  '$state', '$stateParams', '$scope', '$http', 'ThRepositoryModel', 'ThResultSetModel',
+  function CompareChooserCtrl($state, $stateParams, $scope, $http,
                               ThRepositoryModel, ThResultSetModel) {
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
       $scope.originalProject = $scope.newProject = projects[0];
+      $scope.originalTipList = [];
+      $scope.newTipList = [];
+      var getRevisionTips = function(projectName, list) {
+        // due to we push the revision data into list,
```

Comment:
nit: in this added code we are mixing 2 and 4 space indentation.  line 17 is 2 space indentation, then on line 21 we do 4 space indentation.